### ISSUE TITLE
Certain BGS attributes (date_of_death) aren't being saved to the DB when there are calls to BGS

### ISSUE THREAD

**Issue by bgantick** [department-of-veterans-affairs/caseflow#15861]
## Description
There are several instances of veteran records where `Veteran.date_of_death` is not being saved in Caseflow's DB even though calling `date_of_death` returns a date. This is potentially because Caseflow isn't saving veteran records to the DB every time there is a BGS call that gets new data.

### How to recreate in dev environment:
1. In rails console, pull up a veteran that doesn't currently have a `date_of_death`:
```
v = Veteran.where(:file_number => 152003980).first
```
2. Update that veteran's `date_of_death` in `FAKES`:
```
Fakes::BGSService.edit_veteran_record(152003980, :date_of_death, "01/02/2020")
```
3. Confirm disparity between `FAKES` and Caseflow's DB:
```
v.date_of_death
=> Thu, 02 Jan 2020

v
=> ...
 date_of_death: nil
...
```
4. You can also check the Caseflow UI to see that the FNOD badge, FNOD banner, and veteran date of death in Veteran Details are showing:
<img width="678" alt="Screen Shot 2021-02-02 at 4 05 35 PM" src="https://user-images.githubusercontent.com/640182/106662578-8cecbe00-6570-11eb-9c69-3c1aa8982072.png">
<img width="1177" alt="Screen Shot 2021-02-02 at 4 05 46 PM" src="https://user-images.githubusercontent.com/640182/106662585-8f4f1800-6570-11eb-8e7e-0fe05169898e.png">

## Impact

Issue was reported as a result of investigation of a question from the reporting team, as that team relies on information from Caseflow. Related [Slack thread](https://dsva.slack.com/archives/C01CQA12FL7/p1612278232027300).

## Acceptance criteria
- [ ] BGS calls should save results to Caseflow when there is a difference in data, as BGS is the system of record for veteran information

**Comment by ChatthaD** [comment]
@bgantick @yoomlam any update on this would be appreciated. Thanks

**Comment by bgantick** [comment]
@ChatthaD This work is being scheduled into an upcoming sprint.

**Comment by ChatthaD** [comment]
@bgantick Thanks a lot.

**Comment by lomky** [comment]
[what is this chart?](https://github.com/department-of-veterans-affairs/caseflow/issues/11533)
```bash
1 | 
2 | ||
3 | ||||||||||
5 | 
8 | 
```

#### Why 2?
* Optimism
* Fundamentally simple task, except for all the stuff in "Why 3?"

#### Why 3?
* Reviewing lots of locations of code
* Investigate & simple fix
* Update tests
* Unfamiliarity with relevant section of code
* Ensure everywhere we are using DoD is in fact being updated properly

**Comment by ChatthaD** [comment]
> @ChatthaD This work is being scheduled into an upcoming sprint.

Please let us know when this is released & available..

**Comment by bgantick** [comment]
@ChatthaD A fix for this has been deployed (#15990)
